### PR TITLE
Add block weight

### PR DESCRIPTION
**Pull request by imstar15**
image

### CODE REVIEW

File: src/mappings/eventHandlers.ts [hunk 1]
```diff
@@ -74,12 +77,19 @@ async function findOrCreateBlock(substrateBlock: SubstrateBlock): Promise<String
     return existingBaseBlock.id;
   }
 
+  // Calculate weight
+  let blockWeight = new BN(0);
+  events.forEach((event) => {
+    const { event: { method, section, data } } = event;
+    const weight = section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method) ? ((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight : new BN(0);
```
Comment (justinzhou93) [inline src/mappings/eventHandlers.ts]: would this calculation be accurate? it seems like you're only doing weight calculations off of events emitted from extrinsics. There may be many things that happen where the chain doesn't really emit events. So just wanted to check to make sure whether this captures the true weight of the block. I feel like at least all of the block overhead wouldn't be here, as well as all any extrinsics/actions on the block that don't emit an event. unless i'm misunderstanding what an event is (my understanding: see picture below)

image

Comment (imstar15) [reply src/mappings/eventHandlers.ts]: Yes, I think this calculation method is correct.
I looked at the code of polkadot.js apps.

image

The block weight component above uses the `weight` value.

https://github.com/polkadot-js/apps/blob/6267b4a8d3e1fb2c2175017da1b7cc14bb72fea8/packages/page-explorer/src/BlockInfo/Summary.tsx#L70

The calculation of `weight` comes from the following codes.

https://github.com/polkadot-js/apps/blob/6267b4a8d3e1fb2c2175017da1b7cc14bb72fea8/packages/page-explorer/src/BlockInfo/Summary.tsx#L22

```
const [deposits, transfers, weight] = useMemo(
    () => extractEventDetails(events),
    [events]
  );
  
function extractEventDetails (events?: KeyedEvent[]): [BN?, BN?, BN?] {
  return events
    ? events.reduce(([deposits, transfers, weight], { record: { event: { data, method, section } } }) => [
      .....,
      section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method)
        ? weight.iadd(((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight)
        : weight
    ], [new BN(0), new BN(0), new BN(0)])
    : [];
}
```

File: src/mappings/eventHandlers.ts [hunk 2]
```diff
@@ -74,12 +77,19 @@ async function findOrCreateBlock(substrateBlock: SubstrateBlock): Promise<String
...
+    blockWeight.iadd(weight);
+  })
+
   const blockAttributes = {
-    id: id,
+    id, specVersion, timestamp,
```
Comment (justinzhou93) [inline src/mappings/eventHandlers.ts]: nit: format so that it's one field per line, pls!

File: src/mappings/eventHandlers.ts (hunk 1 above)
Comment (justinzhou93) [reply src/mappings/eventHandlers.ts]: ah, makes sense, thanks!

Comment (justinzhou93) [inline src/mappings/eventHandlers.ts]: mind breaking this up into a separate weight calculation function where you pass in the event? This 1 liner is really confusing to read, so it's probably better if you swapped it out for some if statements and good variable naming

**Review by justinzhou93 — APPROVED** [review]
couple of nit picks, otherwise lgtm!

File: src/mappings/eventHandlers.ts (hunk 1 above)
Comment (imstar15) [reply src/mappings/eventHandlers.ts]: Fixed!

File: src/mappings/eventHandlers.ts (hunk 2 above)
Comment (imstar15) [reply src/mappings/eventHandlers.ts]: Fixed!

**Pull request merged**